File: Loenn/entities/synced_zip_mover_activation_controller.lua
local syncedZipMoverActivationController = {}

syncedZipMoverActivationController.name = "CommunalHelper/SyncedZipMoverActivationController"
syncedZipMoverActivationController.depth = -1000000
syncedZipMoverActivationController.texture = "objects/CommunalHelper/syncedZipMoverActivationController/syncedZipMoverActivationController"
syncedZipMoverActivationController.fieldInformation = {
    colorCode = {
        fieldType = "color"
    },
    zipMoverSpeedMultiplier = {
        minimumValue = 0.0
    }
}

syncedZipMoverActivationController.placements = {
    {
        name = "controller",
        data = {
            colorCode = "ff00ff",
            zipMoverSpeedMultiplier = 1.0
        }
    }
}

return syncedZipMoverActivationController

File: Ahorn/entities/misc/SyncedZipMoverActivationController.jl
module CommunalHelperSyncedZipMoverActivationController

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/SyncedZipMoverActivationController" Controller(
    x::Integer,
    y::Integer,
    colorCode::String="ff00ff",
    zipMoverSpeedMultiplier::Number=1.0,
)

const placements = Ahorn.PlacementDict(
    "Synced Zip Mover Activator (Communal Helper)" => Ahorn.EntityPlacement(
        Controller,
    ),
)

const sprite = "objects/CommunalHelper/syncedZipMoverActivationController/syncedZipMoverActivationController"

function Ahorn.selection(entity::Controller)
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Controller)
    Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end
